tests/test_project_legacy_api.py

```diff
@@ -79,22 +74,6 @@ def submit_user(self, env, _id, operations, **kwargs):
         return env.submit(js, _id=_id)
 
 
-class LegacyMockProject(MockProject):
-
-    def write_script_header(self, script, **kwargs):
-        super(LegacyMockProject, self).write_script_header(script, **kwargs)
-
-
-def expect_deprecation_warnings(func):
-
-    def wrapper(self):
-        if six.PY2:
-            raise unittest.SkipTest("Not implemented for Python 2.7.")
-        with self.assertWarns(DeprecationWarning):
-            return func(self)
-    return deprecation.fail_if_not_removed(wrapper)
```

Comment:
I think I'm not understanding how this code works. According to those docs, the decorator raises an `AssertionError` if the test uses code that returns an `UnsupportedWarning`. However, the tests that I removed that were using the `fail_if_not_removed` decorator will fail with an `AttributeError` because there is no longer a `script` method defined for subclasses of `ComputeEnvironment`. Did you want to modify the `ComputeEnvironment` class (and any others with removed code) to raise these warnings when the relevant function is called, and then decorate all of the corresponding tests with `@unittest.expected_failure` so that we can confirm that the tests fail?

Reply:
My general understanding is that you should increase the version number first and then use the unit tests to guide you in which functions *should* be removed.

So I just ran a local test, where I just checked out `develop` and increased the version number to `0.8.0` and I got a bunch of expected failures, but none of the unit tests raised any errors. This  might mean that our coverage is too low or that I misunderstand how this is supposed to work.

Reply:
I don't think the issue is that our coverage is too low. I'm beginning to understand this a little better. You are correct that the expected use-case is that the relevant tests should raise warnings if deprecated APIs are still being used after version bumps. However, I think that the expected next step is that the tests would replace the deprecated code in the unit test with the newer API (and of course also delete the deprecated code paths in the code). In our case, I think we want to remove those deprecated tests entirely because they're superseded by the corresponding tests in the `test_project.py` API. Our tests already cover both APIs, and we check them directly rather than using them as part of more expansive, less "unit"-like tests, so I don't think the only result of trying to keep and fix the expected tests would be duplicating the tests in `test_project.py`. Correct me if you have a different understanding.

Reply:
Yes, I think you are supposed to keep the tests, once any of them runs into an `AssertionError`, because of the version bump you remove corresponding code paths and either update or remove the tests.

I'm still a bit surprised that none of our legacy unit tests ran into any such assertion error on the `develop` branch.